### ISSUE TITLE
[ F# ] character deletion results in strange cursor placement

### ISSUE THREAD

**Issue by dsyme** [mono/monodevelop#3898]
@nosami when using the alpha channel Monodevelop I noticed a very strange movement of the cursor after pressing delete.

Given this code in a file

```fsharp
namespace MonoDevelop.FSharp

module ParsedDocument =
    let f x =
        aaaaa
       |> bbbbb
```

place the cursor in the whitespace before the ``|>``.  Then press delete.  The cursor jumps to after the ``|>``!  

This happen reliably on _any_ line that starts with a `|` character.

This was in a build of MS that included your change #3856.  I haven't checked if that change is responsible, though my understanding was that it didn't involve the delete-key.

> VS bug #569282

**Comment by nosami** [comment]
Confirmed.

**Comment by dsyme** [comment]
@nosami I confirmed that #3856 is the change responsible.  

That said, it was late when I did that and maybe I made a mistake, but I'm pretty sure that's the one

**Comment by nosami** [comment]
Yeah, I started to take a look at it today... I also confirmed the fix on #3856 caused the issue. Just trying to work out how to fix.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #3925 merged by: Jason Imison  
 Author: mono  
 Commit: 6c3a48d3fc4e812259f3d71018fc35ebdc8f765f (xamarin/monodevelop)
